pkg/infrastructure/operator.go

```diff
@@ -54,3 +55,42 @@ func CheckOperatorExists(kubeCli *client.Client, namespace, operatorName string)
 
 	return true, nil
 }
+
+// CheckOperatorExistsUsingSubscription checks whether operator is existing and running. For this check informations from subscription are used.
+// If it is existing but not running, it returns true and an error
+func CheckOperatorExistsUsingSubscription(kubeCli *client.Client, namespace, subscriptionName string) (bool, error) {
```

Comment:
That one seems really nice and more generic :)
I would suggest we keep the CheckKogitoOperatorExists as it was before (means code from CheckOperatorExists would be set back to CheckKogitoOperatorExists). That one would be used for all other operators.
Or even use it for Kogito operator ? @ricardozanini wdyt ?

Reply:
This code can be used to detect operators deployed using OLM.
In case operator is deployed using YAML files then there is no Subscription to carry the informations.
Current implementation is limited to single namespace subscriptions, for global subscriptions the adjustments need to be investigated.

Reply:
ok you are right, forgot this option ...
good for me then